jcl/src/com.ibm.management/share/classes/com/ibm/lang/management/internal/DiagnosticsMXBeanImpl.java

```diff
@@ -0,0 +1,150 @@
+/*[INCLUDE-IF Sidecar17]*/
+/*******************************************************************************
+ * Copyright (c) 2017, 2017 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+ * or the Apache License, Version 2.0 which accompanies this distribution and
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+ *******************************************************************************/
+
+
+package com.ibm.lang.management.internal;
+
+import javax.management.MalformedObjectNameException;
+import javax.management.ObjectName;
+
+import com.ibm.jvm.Dump;
+import com.ibm.jvm.DumpConfigurationUnavailableException;
+import com.ibm.jvm.InvalidDumpOptionException;
+
+import com.ibm.jvm.Trace;
+import com.ibm.lang.management.DiagnosticsMXBean;
+
+public class DiagnosticsMXBeanImpl implements DiagnosticsMXBean {
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public void resetDumpOption() throws DumpConfigurationUnavailableException {
+		Dump.resetDumpOptions();
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public void configureDumpOptions(String dumpOptions) throws InvalidDumpOptionException, DumpConfigurationUnavailableException {
+		System.out.println("Dump Options = " + dumpOptions);
+		Dump.setDumpOptions(dumpOptions);
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public void triggerDump(String dumpAgent) throws IllegalArgumentException {
+                switch (dumpAgent) {
```

Comment:
I think there should be a security check here. Although these Dump APIs contain security checks, they are only enabled when the com.ibm.jvm.enableLegacyDumpSecurity system property is set to true, which it is not by default. Considering you can attach to a running JVM and invoke this, the proper security checks need to be in place.